nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -69,16 +100,164 @@ static NFCSTATUS phNciNfc_Init(void *pContext)
+                                     Offset < pNciContext->RspBuffInfo.wLen &&
+                                     wStatus == NFCSTATUS_SUCCESS; bCount++)
+                    {
+                        /* Parse the buffer */
+                        wStatus = phNciNfc_TlvUtilsGetNxtTlv(&tTlvInfo, &bType, &bLen, &pValue);
+                        Offset += bLen + 2;
+                    }
+
+                    if (wStatus == NFCSTATUS_SUCCESS && pInitRsp->DataHCIPktPayloadLen > 0)
+                    {
+                        /*Note: The HciContext will be created once back in phLibNfc_InitializeProcess*/
+                        wStatus = phNciNfc_CreateConn(UNASSIGNED_DESTID, phNciNfc_e_NFCEE);
+                        if (wStatus == NFCSTATUS_SUCCESS)
+                        {
+                            wStatus = phNciNfc_UpdateConnInfo(UNASSIGNED_DESTID, phNciNfc_e_NFCEE,
+                                CONNHCITYPE_STATIC,
+                                pInitRsp->DataHCINumCredits,
+                                pInitRsp->DataHCIPktPayloadLen);
+
+                            if (wStatus == NFCSTATUS_SUCCESS &&
+                                pInitRsp->DataHCIPktPayloadLen > 0 &&
+                                pInitRsp->DataHCINumCredits == 0)
+                            {
+                                wStatus = phNciNfc_RegForConnCredits(CONNHCITYPE_STATIC,
+                                    &phNciNfc_DelayForCreditNtfCb,
+                                    pNciContext, PHNCINFC_MIN_WAITCREDIT_TO);
+                            }
+                        }
+                    }
+
+                    if (wStatus == NFCSTATUS_SUCCESS)
+                    {
+                        wStatus = phNciNfc_CoreIfSetMaxCtrlPacketSize(&(pNciContext->NciCoreContext),
+                            pInitRsp->CntrlPktPayloadLen);
+                    }
+                }
+                else
+                {
+                    PH_LOG_NCI_CRIT_STR("Invalid number of supported Rf interfaces");
+                    wStatus = NFCSTATUS_FAILED;
+                }
+            }
+            else
+            {
+                wStatus = NFCSTATUS_FAILED;
+            }
+        }
+        else
+        {
+            wStatus = NFCSTATUS_FAILED;
+        }
+    }
+    else
+    {
+        wStatus = NFCSTATUS_INVALID_PARAMETER;
+    }
+    PH_LOG_NCI_FUNC_EXIT();
+    return wStatus;
+}
+
+static NFCSTATUS phNciNfc_ProcessInitRspNci1x(void *pContext, NFCSTATUS Status)
+{
+    NFCSTATUS wStatus = Status;
     phNciNfc_sInitRspParams_t *pInitRsp = NULL;
     pphNciNfc_Context_t pNciContext = pContext;
+    uint8_t *pBuff;
     uint8_t Offset = 0;
     uint16_t wTemp = 0;
 
     PH_LOG_NCI_FUNC_ENTRY();
     if(NULL != pNciContext)
     {
         pInitRsp = &(pNciContext->InitRspParams);
-        if (pNciContext->RspBuffInfo.wLen >= 19)
+        if (pNciContext->RspBuffInfo.wLen >= PH_NCINFC_MIN_CORE_RESET_RSP_LEN)
```

Comment:
Yes, (<data_before_NoOfRfIfSuprt\count(Offset++)> + pInitRsp->NoOfRfIfSuprt + <data_after\count(Offset++)> + PHNCINFC_CORE_MANUF_INFO_LEN_NCI1x <= RspBuffInfo.wLen). Basically to validate input array size and be protected from any possible regressions from future development.

By mistake in naming PH_NCINFC_MIN_CORE_RESET_RSP_LEN you mean creating two different PH_NCINFC_MIN_CORE_INIT_RSP_LEN_xxx for 1x and 2x?